lib/Associations/Many.js

```diff
@@ -369,17 +370,13 @@ function extendInstance(Model, Instance, Driver, association, opts, createInstan
     value: function () {
       var Associations = [];
       var opts = {};
-      var cb = noOperation;
+      var next = noOperation;
 
       var run = function () {
-        var savedAssociations = [];
-        var saveNextAssociation = function () {
-          if (Associations.length === 0) {
-            return cb(null, savedAssociations);
-          }
+        var saveAssociation = function (Association, cb) {
+          const hookOpts = Object.keys(association.props).length > 0 ? opts : undefined;
 
-          var Association = Associations.pop();
```

Comment:
This was the offending line. Rather than switch to `.shift` I decided to simplify by leveraging `async.eachSeries`.